Classes/Private/Manager/Audio/OCTAudioEngine.m

```diff
@@ -0,0 +1,237 @@
+//
+//  OCTAudioEngine.m
+//  objcTox
+//
+//  Created by Chuong Vu on 5/24/15.
+//  Copyright (c) 2015 dvor. All rights reserved.
+//
+
+#import "OCTAudioEngine.h"
+@import AVFoundation;
+
+static const AudioUnitElement kInputBus = 1;
+static const AudioUnitElement kOutputBus = 0;
+
+@interface OCTAudioEngine ()
+
+@property (nonatomic, assign) AUGraph processingGraph;
+@property (nonatomic, assign) AUNode ioNode;
+@property (nonatomic, assign) AudioUnit ioUnit;
+@property (nonatomic, assign) AudioComponentDescription ioUnitDescription;
+
+@end
+
+@implementation OCTAudioEngine
+
+#pragma mark - LifeCycle
+-(instancetype)init
+{
+    self = [super init];
+    if (!self)
+        return nil;
+
+    _ioUnitDescription.componentType = kAudioUnitType_Output;
+    _ioUnitDescription.componentSubType = kAudioUnitSubType_VoiceProcessingIO;
+    _ioUnitDescription.componentManufacturer = kAudioUnitManufacturer_Apple;
+    _ioUnitDescription.componentFlags = 0;
+    _ioUnitDescription.componentFlagsMask = 0;
+
+    NewAUGraph(&_processingGraph);
+    NewAUGraph(&_processingGraph);
+
+    AUGraphAddNode(_processingGraph,
+                   &_ioUnitDescription,
+                   &_ioNode);
+
+    AUGraphOpen(_processingGraph);
+
+    AUGraphNodeInfo(_processingGraph, _ioNode, NULL, &_ioUnit);
+    return self;
+}
+
+#pragma mark - Audio Controls
+-(BOOL)startAudioFlow:(NSError **)error
+{
+    if (![self startAudioSession:error] ||
+        ![self microphoneInput:YES error:error] ||
+        ![self setUpStreamFormat:error] ||
+        ![self initializeGraph:error] ||
+        ![self startGraph:error]) {
+        return NO;
+    }
+
+    return YES;
```

Comment:
It would be even more readable to change it like that:
```
return [self startAudioSession:error] &&
       [self microphoneInput:YES error:error] &&
       [self setUpStreamFormat:error] &&
       [self initializeGraph:error] &&
       [self startGraph:error];
```